### PR TITLE
Cr/26565 follow http redirect

### PR DESCRIPTION
**Pull request by andipunz**
Addresses issue [TS-26565](https://cqse.atlassian.net/browse/TS-26565)

- [ ] Changes are tested adequately
- [ ] Agent's README.md updated in case of user-visible changes
- [ ] CHANGELOG.md updated
- [ ] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)
- [ ] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [ ] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: teamscale-client/build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ publishAs {
 
 dependencies {
     api 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.3'
```
Comment (DreierF) [inline teamscale-client/build.gradle:15]: Was that the only necessary change to fix that? 😅

File: system-tests/http-redirect-test/src/test/java/com/teamscale/client/HttpRedirectSystemTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	private static final int FAKE_TEAMSCALE_PORT = 65438;
+	private static final int FAKE_REDIRECT_PORT = 65437;
+	private static final int AGENT_PORT = 65436;
+
+	@Test
+	public void systemTest() throws Exception {
+		System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StdErrLog");
+		System.setProperty("org.eclipse.jetty.LEVEL", "OFF");
+
+		RedirectMockServer redirectMockServer = new RedirectMockServer(FAKE_REDIRECT_PORT, FAKE_TEAMSCALE_PORT);
```
[the remark this answers is not in the document]
Comment (andipunz) [reply system-tests/http-redirect-test/src/test/java/com/teamscale/client/HttpRedirectSystemTest.java:35]: Sounds like a good idea. I will add code to stop the server for all system tests.

File: system-tests/http-redirect-test/src/test/java/com/teamscale/client/RedirectMockServer.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+public class RedirectMockServer {
+
+	public RedirectMockServer(int port, int redirectTo){
+		Service service = Service.ignite();
+		service.port(port);
+		service.post("/*", (Request request, Response response) -> {
+			String url = request.url();
+			url = url.replace(Integer.toString(port),Integer.toString(redirectTo));
+			response.redirect(url,307);
```
[the remark this answers is not in the document]
Comment (andipunz) [reply system-tests/http-redirect-test/src/test/java/com/teamscale/client/RedirectMockServer.java]: Should be fixed now

File: teamscale-client/build.gradle (hunk 1 above)
Comment (andipunz) [reply teamscale-client/build.gradle:15]: The functionality was added somewhere around version 4.6 of okhttp. Retrofit is still using some 3.x version of okhttp by default. So after some research i found out that it should work with just adding the dependency and it worked.

File: system-tests/http-redirect-test/build.gradle [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+			"teamscale-project=p,teamscale-partition=part,teamscale-commit=master:12345," +
+			"includes=**"
+}
+
+tasks.test.dependsOn(":agent:shadowJar")
+
+dependencies {
+	testImplementation project(':teamscale-client')
+
+	testImplementation 'com.sparkjava:spark-core:2.9.3'
```
[the remark this answers is not in the document]
Comment (andipunz) [reply system-tests/http-redirect-test/build.gradle:23]: Sounds like a valid point. Therefore i would need a little input on how to archive this.

[the remark this answers is not in the document]
Comment (DreierF) [reply system-tests/http-redirect-test/build.gradle:23]: Sure. Could you setup a short meeting in the afternoon to discuss the details.

**Review by DreierF — APPROVED** [review]
LGTM

**Pull request merged**